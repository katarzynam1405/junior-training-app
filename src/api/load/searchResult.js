export default function apiLoadSearchResult() {
    return new Promise((resolve, reject) => {
        setTimeout(() => resolve({
            'Played': 0,
            'Won': 0,
            'Drawn': 0,
            'Lost': 0,
            'Goal For': 0,
            'Goal Against': 0
        }), 700);
    })
}
