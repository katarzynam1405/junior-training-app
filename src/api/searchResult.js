export default function searchResult() {

    const resultList = {
        'Played': 0,
        'Won': 0,
        'Drawn': 0,
        'Lost': 0,
        'Goal For': 0,
        'Goal Against': 0
    };
    return resultList;
}
