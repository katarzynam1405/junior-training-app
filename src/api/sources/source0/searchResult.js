import fetchRequest from './fetchRequest';

export default function apiSrc0SearchResult(token, url) {
  const parser = i => {
    let goalFor = 0;
    let goalAgainst = 0;
    let points = 0;
    let homeTeam = i.result.goalsHomeTeam;
    let awayTeam = i.result.goalsAwayTeam;

    if (i.homeTeamName !== 'Manchester United FC') {
      homeTeam = i.result.goalsAwayTeam;
      awayTeam = i.result.goalsHomeTeam;
    }
    goalFor += homeTeam;
    goalAgainst += awayTeam;

    if (homeTeam > awayTeam) {
      points += 3;
    }
    else if (goalFor == awayTeam) {
      points += 1;
    }
    else {
      points += 0;
    }

    return { goalFor, goalAgainst, points };
  };

  return params =>
    fetchRequest(token, url, params, json => json.fixtures
      .map(parser)
      .reduce((accum, curr) => {
        accum['Goal For'] += curr.goalFor;
        accum['Goal Against'] += curr.goalAgainst;
        accum['Points'] += curr.points;
        return accum;
      }, {
        'Played': 'todo',
        'Points': 0,
        'Won': 'todo',
        'Drawn': 'todo',
        'Lost': 'todo',
        'Goal For': 0,
        'Goal Against': 0
      }) // TODO: Parser (include all stats)
    );
}
