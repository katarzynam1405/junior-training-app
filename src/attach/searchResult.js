export default function attachSearchResult(table, select) {
    document.querySelector(`[${select.outAttr}]`).innerHTML = table;
}
