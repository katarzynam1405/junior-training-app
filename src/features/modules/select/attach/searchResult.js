import ReactDOM from 'react-dom';
/**
 * add table as string to HTML structure
 * @param {string} table
 * @param {Object} select
 */

export default function attachSearchResult(table, select) {
    ReactDOM.render(table, document.querySelector(select.outAttr));
}
