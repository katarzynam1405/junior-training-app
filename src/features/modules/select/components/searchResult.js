// eslint-disable-next-line
import React from 'react';
/**
 * generates table with results in html
 * @param {Object} results object that contains table elements
 * @return string with html table tags that contains searched results
 */

export default function TableRows(results) {
    const headers = Object.keys(results).map(key =>
        <th>{key}</th>
    );
    const values = Object.keys(results).map(key =>
        <td>{results[key]}</td>
    );
    return (
        <table>
            <tr>{headers}</tr>
            <tr>{values}</tr>
        </table>
    );
}
