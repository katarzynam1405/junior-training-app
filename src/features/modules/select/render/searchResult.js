/**
 * generates table with results in html
 * @param {Object} results object that contains table elements
 * @return string with html table tags that contains searched results
 */

export default function renderSearchResult(results) {
    const headers = Object.keys(results).map(key => `<th>${key}</th>`);
    const values = Object.keys(results).map(key => `<td>${results[key]}</td>`);

    return `<thead><tr>${headers.join('')}</tr></thead>` +
        `<tbody><tr>${values.join('')}</tr></tbody>`;
}
