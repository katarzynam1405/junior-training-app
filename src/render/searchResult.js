export default function renderSearchResult(resultList) {
  const headers = [];
  const values = [];

  for (const e in resultList) {
    headers.push(`<th>${e}</th>`);
    values.push(`<td>${resultList[e]}</td>`);
  }

  return `<thead><tr>${headers.join('')}</tr></thead>` +
    `<tbody><tr>${values.join('')}</tr></tbody>`;
}
