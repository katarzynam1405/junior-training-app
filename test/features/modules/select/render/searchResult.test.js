import searchResult from '../../../../../src/features/modules/select/render/searchResult';

describe('f/m/select/render/searchResult', () => {
    it('should math to Snapshot', () => {
        expect(searchResult({ h: 'b'})).toMatchSnapshot()
    });
});
