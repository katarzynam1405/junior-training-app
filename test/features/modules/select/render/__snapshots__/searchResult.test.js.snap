// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`f/m/select/render/searchResult should math to Snapshot 1`] = `
<TableRows
  data={
    Object {
      "h": "b",
    }
  }
/>
`;
